### PR TITLE
Fix `Slot` when used in an RSC

### PR DESCRIPTION
**Pull request by jjenzz**
### Description

in RSCs you cannot pass a `ref` prop (makes sense, given it's a client only thing) so React will throw a nasty error to warn about this if the `ref` is not `undefined`. This becomes problematic with `Slot` because it passes the return type of `composeRefs` (a function) to anything it slots onto which means the `ref` is not `undefined` regardless of whether you are attaching one and the error occurs.

This PR ensures the ref is only composed by `Slot` if there is a defined `forwardRef` and allows consumers to build their own slottable comps to use in RSCs e.g.:

```jsx
// span by default
<Text asChild size={'large'}>
  <p>foo</p>
</Text>
```

### CODE REVIEW

File: packages/react/slot/src/Slot.tsx [hunk 1]
```diff
@@ -63,7 +63,7 @@ const SlotClone = React.forwardRef<any, SlotCloneProps>((props, forwardedRef) =>
   if (React.isValidElement(children)) {
     return React.cloneElement(children, {
       ...mergeProps(slotProps, children.props),
-      ref: composeRefs(forwardedRef, (children as any).ref),
+      ref: forwardedRef ? composeRefs(forwardedRef, (children as any).ref) : (children as any).ref,
```
Comment (jjenzz) [inline packages/react/slot/src/Slot.tsx:66]: could also just do this tbh (so it isn't overriding `ref` if it doesn't need to) but lmk which you'd prefer :)

Screenshot 2023-04-27 at 16 21 20

Comment (benoitgrelard) [reply packages/react/slot/src/Slot.tsx:66]: Wouldn't that one be missing `children.ref` then?

Comment (benoitgrelard) [reply packages/react/slot/src/Slot.tsx:66]: I think I prefer the explicit way you have written though either way.

**Review by benoitgrelard — APPROVED** [review]

**Comment by kuus** [comment]
https://github.com/radix-ui/primitives/issues/2011

**Pull request merged**